core/services/vrf/listener_v1.go

```diff
@@ -370,6 +370,7 @@ func (lsn *listenerV1) ProcessRequest(req *solidity_vrf_coordinator_interface.VR
 	})
 
 	run := pipeline.NewRun(*lsn.job.PipelineSpec, vars)
+	// The VRF pipeline has no async tasks, so we don't need to check for `incomplete`
 	if _, err = lsn.pipelineRunner.Run(context.Background(), &run, lsn.l, true, func(tx pg.Queryer) error {
```

Comment:
Does this happen synchronously? I'm wondering if the run errors will already be populated at this point

Reply:
Yep, for VRF this should be synchronous, and each task in the pipeline depends on the previous task finishing completely.